### ISSUE TITLE
Laser Meassure

### ISSUE THREAD

**Issue by LPFraile** [rutian/MyPart#1]
The Laser on the BOM is 35mm length and can not be placed on the 3D printer design box. What is the correct laser?

**Comment by rutian** [comment]
It is the correct laser, unfortunately, the outer black metal shell has to be stripped away in order to get to the laser module itself. This can be done by cutting an opening at the end of the laser casing with a file or hacksaw, then stripping away the casing with needle nose pliers. A new design revision is in the works (which simplifies the laser assembly/alignment), and should be made public in ~1 week.

**Comment by LPFraile** [comment]
Thanks very much for your fast answer. I have another issue. I try to programming the board by connect TX RX and DTR pins to the specific ones on my FOca V2.i2 programmer on 3.3 v. and feed the MyPart board from the mini usb of the board. When i try to program with a simple test of temperature humidity sensor i achieved an erros

"

get

fail.......fail.......fail.......
"

I make something wrong on my way of programming? Any idea of what can be the problem?

Thank you very much on advance

L.P

---

De: rutian notifications@github.com
Enviado: miércoles, 12 de octubre de 2016 08:16 p. m.
Para: rutian/MyPart
Cc: LPFraile; Author
Asunto: Re: [rutian/MyPart] Laser Meassure (#1)

It is the correct laser, unfortunately, the outer black metal shell has to be stripped away in order to get to the laser module itself. This can be done by cutting an opening at the end of the laser casing with a file or hacksaw, then stripping away the casing with needle nose pliers. A new design revision is in the works (which simplifies the laser assembly/alignment), and should be made public in ~1 week.

##

**Comment by rutian** [comment]
Hard to say... but I would check to see if gnd is connected, and that there is a .1uF capacitor between the DTR pin of your programmer, and the reset pin of the rfduino. 3.3V should not be connected because the voltage regulators on that board are not reverse current protected. 

Let me know if that helps!

**Comment by LPFraile** [comment]
Hello

I have include it the capacitor between DTR pin and RESET pin and i achieved the same result

"Timeout reading from uart.

fail.......fail.......fail......."

I have check and the connexions are ok and the RF duino is well feed with 3V from the USB. But the SDA, RESET and SCL connected with the FTDI programmer is 3.3 V, may be this the problem. May you know something else that i can try?

Thank you very much for your help

Best regards

LP

---

De: rutian notifications@github.com
Enviado: lunes, 17 de octubre de 2016 07:00 p. m.
Para: rutian/MyPart
Cc: LPFraile; Author
Asunto: Re: [rutian/MyPart] Laser Meassure (#1)

Hard to say... but I would check to see if gnd is connected, and that there is a .1uF capacitor between the DTR pin of your programmer, and the reset pin of the rfduino. 3.3V should not be connected because the voltage regulators on that board are not reverse current protected.

Let me know if that helps!

##

**Comment by LPFraile** [comment]
Hello

Sorry agin but you think that if i bay the RFD22121 USB Programme I will  be able to programming the chip?

Regards and thank you on advance

---

De: rutian notifications@github.com
Enviado: lunes, 17 de octubre de 2016 07:00 p. m.
Para: rutian/MyPart
Cc: LPFraile; Author
Asunto: Re: [rutian/MyPart] Laser Meassure (#1)

Hard to say... but I would check to see if gnd is connected, and that there is a .1uF capacitor between the DTR pin of your programmer, and the reset pin of the rfduino. 3.3V should not be connected because the voltage regulators on that board are not reverse current protected.

Let me know if that helps!

##

**Comment by rutian** [comment]
Yes! With that programmer, you should be able to isolate potential issues. 

Other things to try (if you haven't already) would be to (1) continuity check between all pins coming out of the programmer and the pins on the rfduino they go to (2) making sure that the rfduino is getting power by probing the vcc and gnd pins on the rfduino. 

Also, how are you powering the board? There should be a lipo battery attached. I'm not sure what will happen if 3v is given to the usb port, since that power goes into a lipo charging IC (If there is no battery, it may just detect that, and do nothing).

**Comment by LPFraile** [comment]
Hello

Thank you very much for you help!! I have order a programmer i am expecting to came. All this i have already check and the rfduino if feed ok. As i understand is the  bootloder what i miss. I spouse that i could try programming directly the rfduino overpassing the bootlooder but I just will wait for the programmer. I will lend you know if i achieved to work.

thanks again and best regards

LP

---

De: rutian notifications@github.com
Enviado: miércoles, 19 de octubre de 2016 08:46 a. m.
Para: rutian/MyPart
Cc: LPFraile; Author
Asunto: Re: [rutian/MyPart] Laser Meassure (#1)

Yes! With that programmer, you should be able to isolate potential issues.

Other things to try (if you haven't already) would be to (1) continuity check between all pins coming out of the programmer and the pins on the rfduino they go to (2) making sure that the rfduino is getting power by probing the vcc and gnd pins on the rfduino.

Also, how are you powering the board? There should be a lipo battery attached. I'm not sure what will happen if 3v is given to the usb port, since that power goes into a lipo charging IC (If there is no battery, it may just detect that, and do nothing).

##

**Comment by rutian** [comment]
Hi LP,

Hope everything is going smoothly. The revised design files for the new version of the MyPart have finally been updated. We've also included the design files for a test chamber that can be used to run automated tests for calibration. There are some build instructions for the chamber [here](http://www.instructables.com/id/How-to-Build-a-Test-Chamber-for-Air-Particle-Senso/) if you are interested.

**Comment by LPFraile** [comment]
Thank you very much!! I am waiting yet for the programmer to came. When i will have new i will lend it you know.

Regards

LP

---

De: rutian notifications@github.com
Enviado: martes, 8 de noviembre de 2016 09:57 a. m.
Para: rutian/MyPart
Cc: LPFraile; Author
Asunto: Re: [rutian/MyPart] Laser Meassure (#1)

Hi LP,

Hope everything is going smoothly. The revised design files for the new version of the MyPart have finally been updated. We've also included the design files for a test chamber that can be used to run automated tests for calibration. There are some build instructions for the chamber herehttp://www.instructables.com/id/How-to-Build-a-Test-Chamber-for-Air-Particle-Senso/ if you are interested.

##

**Comment by LPFraile** [comment]
Hello
I am again on a try to make the mypart device working. But i understand that i have the older version board. Is any new aspect on the new board to be taking account? Also  I would like to know where I can find the RFduino firmware  for read particles and send it on BLE. Now i have the RFDuino programmer and the   LEd test is look correct.
Thanks on advances

**Comment by LPFraile** [comment]
But when i try to run the for_test_chamber scket I achieved the follow compile error. Have you any idea what can be wrong? 

Arduino: 1.8.2 (Linux), Board: "RFduino"
arm-none-eabi-g++: error: core/syscalls.c.o: No such file or directory
exit status 1
Error compiling for board RFduino.

**Comment by rutian** [comment]
Hello, 
There's a few differences between the newer and older board: (1) physical size (2) different gpio expander IC (3) different humidity/temperature sensor (4) a single rgb led instead of the multi-led array. (5) the button can now be triggered through the programming port, which is how the sensors can be triggered remotely for the for_test_chamber.ino sketch. There might be more, but those are the main ones that I can think of. You will need the libraries for the two new I2C devices: https://github.com/rutian/MyPart/tree/master/testing_chamber/testing_scripts/custom_arduino_libraries.


as for the compile error, I'm not sure what the problem is. It could be something about how the libraries for the rfduino are stored/configured?
